### ISSUE TITLE
Gc shows that some repos are damaged, but fsck does not repair them

### ISSUE THREAD

**Issue by n3tdom** [haiwen/seafile#1884]
There was a similiar issue year ago: #1547 GC says repo is corrupted but cannot be fixed by multiple fsck run

We are running Seafile server 6.0.7, ~1TB of data
command log:
	gc shows errors:
	./seaf-gc.sh
	[04/16/17 12:07:49] gc-core.c(456): === GC is finished ===
	[04/16/17 12:07:49] gc-core.c(460): The following repos are damaged. You can run seaf-fsck to fix them.
	[04/16/17 12:07:49] gc-core.c(463): 0e6fc2c2-eb5a-4998-82fe-c11f75289315
	[04/16/17 12:07:49] gc-core.c(463): 1f0417d4-ef9a-4b58-b7b2-2f7af627f63a
	[04/16/17 12:07:49] gc-core.c(463): c7b10142-6d4d-4aa8-afd1-b69e537ed8d5
	[04/16/17 12:07:49] gc-core.c(463): cdfbd7fd-9cf8-43f7-a291-06bd15de45d7
	[04/16/17 12:07:49] gc-core.c(463): f18f52cc-498f-4975-8b16-3ce1fb1e712f
fsck shows no errors:
	./seaf-fsck.sh -r 0e6fc2c2-eb5a-4998-82fe-c11f75289315
	Starting seaf-fsck, please wait ...
	[04/16/17 12:13:05] fsck.c(548): Running fsck for repo 0e6fc2c2-eb5a-4998-82fe-c11f75289315.
	[04/16/17 12:13:05] fsck.c(409): Checking file system integrity of repo Моя библиотека(0e6fc2c2-eb5a-4998-82fe-c11f75289315) ...
	[04/16/17 12:13:16] fsck.c(612): Fsck finished for repo 0e6fc2c2.

We understand that it is a corruption somewhere in history data. 
Disabling -> making changes in library -> enabling history doesn't fix error, gc still thinks history is corrupt.
Creating another library, moving all contents to it, then deleting problem library - fixes the problem.
The downside is, that all permissions and links to library break (obviously), so it is not very convinient solution

Is it possible to determine what caused this kind of corruption?
Will there be some mechanism to repair that kind of error without recreating repos?

**Comment by killing** [comment]
The corruption in history will not affect daily use of Seafile. But when some history data is corrupted, the unused blocks cannot be removed by GC. We can add an option to forcibly remove those blocks. But this is not a priority task for us though.

**Comment by n3tdom** [comment]
Understood. We will appriciate having that option in GC! 
The error occurred only once for us by now, so it's not a big problem yet.

**Comment by n3tdom** [comment]
Understood. We will appriciate having that option in GC! 
The error occurred only once for us by now, so it's not a big problem yet.

**Comment by galvanopus** [comment]
@killing Has this been fixed or you are closing old issues to check whether they are obsolete?

**Comment by killing** [comment]
This is a designed behavior. The damage in history doesn't affect usage. So it's actually not a real issue. So I marked it as closed.

**Comment by galvanopus** [comment]
I'd say this is a real issue as if repo's history is damaged you can't run GC on that repo to eliminate the history error.

**Comment by galvanopus** [comment]
@killing is there any chance to have the issue with GC fixed? It's OK to clear the damaged history, but it is not desirable to move all data to the new library just to fix this issue.
